src/cnd.rs

```diff
@@ -0,0 +1,151 @@
+use rand;
+use rand::Rng;
+use serde::Serialize;
+use std::io::Write;
+use std::net::{IpAddr, Ipv4Addr};
+use std::process::Command;
+use tempfile::{self, TempPath};
+use tokio_process::{Child, CommandExt};
+
+pub struct Cnd {
+    pub settings: Settings,
+    _config_file: TempPath,
+    pub process: Child,
+}
+
+impl Cnd {
+    pub fn start(settings: Settings) -> Cnd {
+        let mut config_file = tempfile::Builder::new().suffix(".toml").tempfile().unwrap();
+        config_file
+            .write(
+                toml::to_string(&settings)
+                    .expect("could not serialize settings")
+                    .as_ref(),
+            )
+            .expect("could not write to temporary file");
+        let config_file = config_file.into_temp_path();
+
+        let process = Command::new("cnd")
+            .arg("--config")
+            .arg(config_file.to_str().unwrap())
+            .stdout(std::process::Stdio::null())
+            .spawn_async()
+            .expect("failed to start btsieve");
+
+        Cnd {
+            settings,
+            _config_file: config_file,
+            process,
+        }
+    }
+}
+
+#[derive(Clone, Debug, Serialize, Default)]
+pub struct Settings {
+    pub comit: Comit,
+    pub network: Network,
+    pub http_api: HttpSocket,
+    pub btsieve: Btsieve,
+    pub web_gui: Option<HttpSocket>,
+}
+
+#[derive(Clone, Debug, Serialize)]
+pub struct Comit {
+    #[serde(with = "hex_serde")]
+    pub secret_seed: [u8; 32],
+}
+
+#[derive(Clone, Debug, Serialize)]
+pub struct Network {
+    pub listen: Vec<String>,
+}
+#[derive(Clone, Debug, Serialize)]
+pub struct HttpSocket {
+    pub address: IpAddr,
+    pub port: u16,
+}
+
+#[derive(Clone, Debug, Serialize)]
+pub struct Btsieve {
+    pub url: String,
+    pub bitcoin: PollParameters,
+    pub ethereum: PollParameters,
+}
+
+#[derive(Clone, Debug, Serialize)]
+pub struct PollParameters {
+    pub poll_interval_secs: u16,
+    pub network: String,
+}
+
+impl Default for Comit {
```

Comment:
Idea: Instead of spreading out the default config values, we could also just have a single `Default` impl of `Settings`.

Reply:
Actually it's easier to allow pick and choosing what should be overridden See in `start_env.rs`:
```
        let settings = cnd::Settings {
            btsieve: cnd::Btsieve {
                url: btsieve_url,
                ..Default::default()
            },
            ..Default::default()
        };
```